### PR TITLE
cli: change webinspector `js-shell` to use convenient classes

### PR DESCRIPTION
**Pull request by guysalt**

### CODE REVIEW

File: pymobiledevice3/cli/webinspector.py [hunk 1]
```diff
@@ -295,3 +210,109 @@ def cdp(lockdown: LockdownClient, host, port):
     udid = lockdown.udid
     uvicorn.run('pymobiledevice3.cli.webinspector:create_app', host=host, port=port, factory=True,
                 ws_ping_timeout=None, ws='wsproto', loop='asyncio')
+
+
+class JsShell(ABC):
```
Comment (matan1008) [inline pymobiledevice3/cli/webinspector.py:217]: add `start`

File: pymobiledevice3/cli/webinspector.py [hunk 2]
```diff
@@ -295,3 +210,109 @@ def cdp(lockdown: LockdownClient, host, port):
...
+        available_pages = list(self.inspector.get_open_pages().get('Safari', []))
+        if not available_pages:
+            logger.error('Unable to find available pages (try to unlock device)')
+            return
+
+        page_query = [inquirer.List('page', message='choose page', choices=available_pages, carousel=True)]
+        page = inquirer.prompt(page_query, theme=GreenPassion(), raise_keyboard_interrupt=True)['page']
+        self.page = page
+
+    async def start_js_shell(self, url: str = ''):
```
Comment (matan1008) [inline pymobiledevice3/cli/webinspector.py]: rename to `start`

File: pymobiledevice3/cli/webinspector.py [hunk 3]
```diff
@@ -295,3 +210,109 @@ def cdp(lockdown: LockdownClient, host, port):
...
+
+    def __init__(self, lockdown: LockdownClient, timeout: float, app: str):
+        super().__init__(lockdown, timeout, app)
+        self.automation_session = None
+        self.driver = None
+
+    async def evaluate_expression(self, exp):
+        return self.driver.execute_script(f'return {exp}')
+
+    def start_js_shell(self, url: str = ''):
```
Comment (matan1008) [inline pymobiledevice3/cli/webinspector.py]: rename to `start`

File: pymobiledevice3/cli/webinspector.py (hunk 2 above)
Comment (matan1008) [reply pymobiledevice3/cli/webinspector.py]: + keep the api similar to the AutomationJsShell

File: pymobiledevice3/cli/webinspector.py [hunk 4]
```diff
@@ -295,3 +210,109 @@ def cdp(lockdown: LockdownClient, host, port):
...
+class JsShell(ABC):
+
+    def __init__(self, lockdown: LockdownClient, timeout: float, app: str):
+        super().__init__()
+        self.inspector = WebinspectorService(lockdown=lockdown)
+        self.inspector.connect(timeout)
+        self.application = self.inspector.open_app(app)
+        self.prompt_session = None
+
+    def start_prompt_session(self):
```
Comment (matan1008) [inline pymobiledevice3/cli/webinspector.py]: merge with the init

File: pymobiledevice3/cli/webinspector.py [hunk 5]
```diff
@@ -295,3 +210,109 @@ def cdp(lockdown: LockdownClient, host, port):
     udid = lockdown.udid
     uvicorn.run('pymobiledevice3.cli.webinspector:create_app', host=host, port=port, factory=True,
                 ws_ping_timeout=None, ws='wsproto', loop='asyncio')
+
+
+class JsShell(ABC):
+
+    def __init__(self, lockdown: LockdownClient, timeout: float, app: str):
```
Comment (matan1008) [inline pymobiledevice3/cli/webinspector.py]: Might be easier the add staticmethod / classmethod that initialize (and selects the page)

File: pymobiledevice3/cli/webinspector.py (hunk 2 above)
Comment (guysalt) [reply pymobiledevice3/cli/webinspector.py]: > rename to `start`

done

Comment (guysalt) [reply pymobiledevice3/cli/webinspector.py]: > * keep the api similar to the AutomationJsShell

what do you mean? with the `try-finally` `inspector.close`?

File: pymobiledevice3/cli/webinspector.py (hunk 3 above)
Comment (guysalt) [reply pymobiledevice3/cli/webinspector.py]: done

File: pymobiledevice3/cli/webinspector.py (hunk 4 above)
Comment (guysalt) [reply pymobiledevice3/cli/webinspector.py]: done

File: pymobiledevice3/cli/webinspector.py (hunk 1 above)
Comment (guysalt) [reply pymobiledevice3/cli/webinspector.py:217]: done

File: pymobiledevice3/cli/webinspector.py (hunk 2 above)
Comment (guysalt) [reply pymobiledevice3/cli/webinspector.py]: done

File: pymobiledevice3/cli/webinspector.py [hunk 6]
```diff
@@ -15,32 +18,19 @@
 from prompt_toolkit.styles import style_from_pygments_cls
 from pygments import highlight, lexers, formatters
 from pygments.styles import get_style_by_name
-
```
Comment (matan1008) [inline pymobiledevice3/cli/webinspector.py]: ?

**Review by matan1008 — APPROVED** [review]

**Pull request merged**